external_resources_io/terraform/generators.py

```diff
@@ -0,0 +1,239 @@
+            return f"set({_get_terraform_type(args[0])})" if args else "set(any)"
+
+        case t if t is dict:
+            return f"map({_get_terraform_type(args[1])})" if args else "map(any)"
+
+        case t if t is Literal:
+            return "string"
+
+        case t if t in {UnionType, Union}:
+            if type(None) in args:
+                return _get_terraform_type(args[0])
+            return "any"
+
+        case _:
+            return "any"
+
+
+def _convert_basic_types(python_type: Any) -> str:  # noqa: PLR0911
+    """Convert basic python types to Terraform types."""
+    match python_type:
+        case t if t is str:
+            return "string"
+        case t if t is int:
+            return "number"
+        case t if t is bool:
+            return "bool"
+        case t if t is list:
+            return "list(any)"
+        case t if t is set:
+            return "set(any)"
+        case t if t is dict:
+            return "map(any)"
+        case t if issubclass(t, BaseModel):
+            # nested model
+            fields_types = ",".join(
+                f"{k} = {v['type']}" for k, v in _generate_fields(t).items()
+            )
+            return f"object({{{fields_types}}})"
+        case _:
+            return "any"
+
+
+def _get_terraform_type(python_type: Any) -> str:
+    """Maps Python types to Terraform types."""
+    origin = get_origin(python_type)
+    args = get_args(python_type)
+
+    return (
+        _convert_generic_types(origin, args)
+        if origin is not None
+        else _convert_basic_types(python_type)
+    )
+
+
+def _convert_json_value_to_hcl(value: Any) -> str:  # noqa: PLR0911
+    """Converts a JSON value to HCL."""
+    match value:
+        case t if isinstance(t, str):
+            return f'"{value}"'
+        case t if isinstance(t, bool):
+            return str(value).lower()
+        case t if isinstance(t, int | float):
+            return str(value)
+        case t if isinstance(t, list | set):
+            if not value:
+                return "[]"
+            return "[" + ",".join(_convert_json_value_to_hcl(e) for e in value) + "]"
+        case t if isinstance(t, dict):
+            if not value:
+                return "{}"
+            pairs = []
+            for k, v in value.items():
+                converted_v = _convert_json_value_to_hcl(v)
+                pairs.append(f"{k} = {converted_v}")
+            return "{\n" + "\n".join(pairs) + "\n}"
+        case None:
+            return "null"
+        case _:
+            return str(value)
+
+
+def _convert_json_to_hcl(data: dict) -> str:
+    variables = data.get("variable", {})
+    hcl_blocks = []
+
+    for var_name in sorted(variables):
+        block_lines = [f'variable "{var_name}" {{']
+        for key, value in variables[var_name].items():
+            hcl_value = value if key == "type" else _convert_json_value_to_hcl(value)
+            block_lines.append(f"{key} = {hcl_value}")
```

Comment:
we only want to sort by var name in variables level, still keep insert order inside var block like `type` first, `default` second, right?
And for old code to support order, this should work 

```python
for var_name, var_config in sorted(variables.items()):
```